### DISCUSSION TITLE
Higher-order interpolation for affine transforms

### DISCUSSION THREAD

**Discussion by samcoveney** [dipy/dipy#2517] [Registration]
The method AffineMap.transform only accepts 'linear' and 'nearest', since only linear and nearest neighbour transforms are implemented. These are implemented at a fairly low level, giving fast speeds for image registration.

It is very common to use linear interpolation for registration even if higher-order (e.g. cubic) interpolation methods are implemented (this is the default in ElastiX). But it is common to apply higher-order interpolation for a final image transform. I don't believe this is currently implemented in DIPY, so I have been thinking...

In dipy/align/imaffine.py we have
  
    _transform_method[(2, 'nearest')] = vf.transform_2d_affine_nn
    _transform_method[(3, 'nearest')] = vf.transform_3d_affine_nn
    _transform_method[(2, 'linear')] = vf.transform_2d_affine
    _transform_method[(3, 'linear')] = vf.transform_3d_affine

Would there be interest in specifying some higher-order interpolation options, by expanding the possible transform methods, but simply borrowing an interpolation method from scipy, or creating one using numpy? While this would probably be very slow for registration, a single call to such a function for the purposes of applying a final transformation would probably be okay, and speed would be less of an issue.

Any thoughts on this?

**Comment by skoudoro** [comment]
+1 for this. It is fine if it is slow for the first version as long as we warn the users.  Also, It permits to show how to do it for some users.

Later on, if there are too many complaints, we can focus and implement a faster version. 

Sorry for the late reply but this is a good idea
